elliptic-curve-crate/src/weierstrass.rs

```diff
@@ -4,8 +4,30 @@ pub mod curve;
 pub mod point;
 pub mod public_key;
 
-pub use curve::{Curve, ScalarBytes};
+pub use curve::Curve;
 pub use point::{
     CompressedCurvePoint, CompressedPointSize, UncompressedCurvePoint, UncompressedPointSize,
 };
 pub use public_key::PublicKey;
+
+use crate::{consts::U1, ScalarBytes};
+use core::ops::Add;
+use generic_array::ArrayLength;
+use subtle::{ConditionallySelectable, CtOption};
+
+/// Fixed-base scalar multiplication
+pub trait FixedBaseScalarMul: Curve
+where
+    <Self::ScalarSize as Add>::Output: Add<U1>,
+    CompressedCurvePoint<Self>: From<Self::Point>,
+    UncompressedCurvePoint<Self>: From<Self::Point>,
+    CompressedPointSize<Self::ScalarSize>: ArrayLength<u8>,
+    UncompressedPointSize<Self::ScalarSize>: ArrayLength<u8>,
+{
+    /// Elliptic curve point type
+    type Point: ConditionallySelectable + Default;
```

Comment:
The `Default` bound is necessitated by the current use of `CtOption`, namely `and_then`/`map`:

https://docs.rs/subtle/2.2.3/subtle/struct.CtOption.html#method.map

I think it'd be nice to get rid of it if possible...

Reply:
Got rid of it 🎉